core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,649 @@
+const textToSpeechToggle = 'label[for="text-speech-switch"]';
+const feedbackToggleOff = 'label[for="feedback-switch"]';
+
+const editbutton = '.e2e-test-edit-roles';
+const addUserName = '#newMemberUsername';
+const addRoleBar = 'mat-select.e2e-test-role-select';
+const collaborator = 'Collaborator (can make changes)';
+const playtester = 'Playtester (can give feedback)';
+const saveRole = 'button.e2e-test-save-role';
+const deleteExplorationButton = 'button.e2e-test-delete-exploration-button';
+const saveDraftButton = 'button.e2e-test-save-changes';
+const publishButton = 'button.e2e-test-publish-exploration';
+const discardDraftButton = 'a.e2e-test-discard-changes';
+const discardConfirmButton = 'button.e2e-test-confirm-discard-changes';
+const deleteConfirmButton = 'button.e2e-test-really-delete-exploration-button';
+const voiceArtistEditButton = 'span.e2e-test-edit-voice-artist-roles';
+const voiceArtistSaveButton = 'button.e2e-test-add-voice-artist-role-button';
+const publishConfirmButton = 'button.e2e-test-confirm-publish';
+const commitMessage = 'textarea.e2e-test-commit-message-input';
+const closePublishedPopUp = 'button.e2e-test-share-publish-close';
+const addVoiceArtistUserName = 'input#newVoicAartistUsername';
+
+/**
+ * For mobile.
+ */
+const navBarOpener = 'i.e2e-test-mobile-options';
+const optionsDropDown = 'div.e2e-test-mobile-options-dropdown';
+const mobileSettingsBar = 'li.e2e-test-mobile-settings-button';
+const basicSettingsDropDown = 'h3.e2e-test-settings-container';
+const feedbackSettingsDropdown = 'h3.e2e-test-feedback-settings-container';
+const permissionSettingsDropDown = 'h3.e2e-test-permission-settings-container';
+const voiceArtistSettingsDropDown =
+  'h3.e2e-test-voice-artists-settings-container';
+const rolesSettingsDropDown = 'h3.e2e-test-roles-settings-container';
+const advanceSettingsDropDown = 'h3.e2e-test-advanced-settings-container';
+const explorationControlsSettingsDropDown =
+  'h3.e2e-test-controls-bar-settings-container';
+const mobileSaveDraftButton = 'button.e2e-test-save-changes-for-small-screens';
+const mobilePublishButton = 'button.e2e-test-mobile-publish-button';
+const publishButtonDropDown = 'div.e2e-test-mobile-changes-dropdown';
+const mobileDiscardButton = 'div.e2e-test-mobile-exploration-discard-tab';
+
+let explorationUrlAfterPublished = '';
+export class ExplorationCreator extends BaseUser {
+  /**
+   * This function helps in reaching dashboard Url.
+   */
+  async openCreatorDashboardPage(): Promise<void> {
+    await this.goto(creatorDashboardUrl);
+  }
+
+  /**
+   * This function helps in reaching editor section.
+   */
+  async switchToEditorTab(): Promise<void> {
+    await this.clickOn(createNewExplorationButton);
+    await this.clickOn(takeMeToEditorButton);
+  }
+
+  /**
+   * This function helps in updating Card Name.
+   */
+  async updateCardName(cardName: string): Promise<void> {
+    await this.clickOn(addCardName);
```

Comment:
I don't know what "addCardName" is, shouldn't it be something like "editCardNameButton"?